src/GenerateProgrammaticPipelineFromConfig/Generator.php

```diff
@@ -56,6 +57,9 @@ class Generator implements Constants
             // Override the following to use an alternate "not found" delegate.
             NotFoundDelegate::class => NotFoundDelegateFactory::class,
```

Comment:
this still causes `An alias by the name "zendexpressivedelegatenotfounddelegate" or "Zend\Expressive\Delegate\NotFoundDelegate" already exists` but don't know if I can remove it or have to change to something else @weierophinney can you help?

Reply:
Where/when do you get that error message, exactly? 

I'm reviewing the slack conversation where you detail these problems, and I'm having trouble understanding how they could happen in the first place. Configuration such as this is _aggregated_ during bootstrapping _before_ seeding the container itself. If you're getting such messages, it's usually an indication that you're _manually_ building your container.

Can you provide a bit more context, please?

Reply:
I found it when helping to fix this https://github.com/Naktibalda/codeception-zend-expressive-tests/issues/7 I was able to fix it by requiring servicemanager v3 but expressive should work with any PSR-7 container implementation and when SM2 I get those errors

Reply:
Okay, then this may be a moot issue.

Expressive v2 releases _do not support zend-servicemanager v2_. They pin to v3 releases (see [here](https://github.com/zendframework/zend-expressive-skeleton/blob/2.0.0/src/ExpressiveInstaller/config.php) and [here](https://github.com/zendframework/zend-expressive-skeleton/blob/2.0.0/composer.json). While Expressive v1 releases supported zend-servicemanager v2, we dropped support for those versions specifically because of discrepancies in how they operate in relation to v3.

If the error only occurs when using zend-servicemanager v2, then we can close this issue; we do not support that version in releases where this tooling would be relevant.

Reply:
It's not a matter of whether or not both are PSR-11 compatible; for that purpose, the `has()` and `get()` methods of each operate identically.

The issue is that zend-servicemanager v2 and v3 have a few differences in how they handle container population, with v3 fixing issues in v2 that we could not address without a BC break. One of these is how it identifies service conflicts during initial configuration. v3 has a better algorithm — but it's one we cannot backport, as it introduces BC breaks elsewhere. For this reason, we chose to only support zend-servicemanager v3 in Expressive v2 and up.